utils/db/setupRepos.ts

```diff
@@ -7,155 +8,52 @@ export interface SetupReposArgs {
     install_id: string;
 }
 
-interface SetupResponse {
-    message: string;
-}
-
-export const saveSetupReposInDb = async (args: SetupReposArgs): Promise<SetupResponse> => {
+export const saveSetupReposInDb = async (args: SetupReposArgs, userId: string): Promise<boolean> => {
     const { repo_owner, repo_provider, repo_names, install_id } = args;
-    const query = `SELECT EXISTS(SELECT 1 FROM repos WHERE repo_owner = $1 AND repo_provider = $2)`
-
-    return conn.query(query, [repo_owner, repo_provider])
-        .then(async (orgExistsResult) => {
-        const orgExists = orgExistsResult.rows[0].exists;
-
-        if (!orgExists) {
-            await addOrganization(repo_owner, repo_provider, repo_names, install_id).catch(error => {
-                console.error(`[saveSetupReposInDb] Could not add organisation in repos table for: ${repo_provider}/${repo_owner} for install_id ${install_id}`, error );
-                throw new Error('Failed to add organization', error);
-            });
-        } else {
-            await handleRepoSetupAndRemoveExtra(repo_names, repo_owner, repo_provider, install_id).catch(error => {
-                console.error(`[saveSetupreposInDb] error in handleReposetupAndRemoveExtra function: ${repo_provider}/${repo_owner} for install_id ${install_id}`, error);
-                throw new Error('Failed to handle repo setup and remove extra repositories', error);
-            });
+    const insertReposQuery = `
+        INSERT INTO repos (repo_name, install_id, repo_owner, repo_provider)
+        SELECT repo_name, ARRAY[$4]::INT[], $1, $2
+        FROM unnest($3::TEXT[]) AS t(repo_name)
+        ON CONFLICT (repo_name) DO UPDATE
+            SET install_ids = CASE
+            WHEN NOT (repos.install_id && ARRAY[$4]::INT[]) -- user's install_id not present
+            THEN repos.install_id || $4 -- append user's install_id
+            ELSE repos.install_id -- no update needed
+            END,
+            repo_owner = $1,
+            repo_provider = $2
+        RETURNING id AS repo_id;
+    `;
+    try {
+        await conn.query('BEGIN');
+        const { rowCount: reposRowCount, rows } = await conn.query(insertReposQuery, [repo_owner, repo_provider, repo_names, install_id]);
+        if (reposRowCount === 0) {
+            await conn.query('ROLLBACK');
```

Comment:
Extremely low LOGAF.

If the updated row count is 0, then does it even matter?

Reply:
Tapish: It doesn't matter, but we will still keep this statement as a good practice.